src/services/notehub/NotehubDataProvider.ts

```diff
@@ -133,7 +133,12 @@ export default class NotehubDataProvider implements DataProvider {
       return {
         macAddress: gatewaySensorInfo.macAddress,
         gatewayUID: gatewaySensorInfo.gatewayUID,
-        ...(sensorNameInfo?.body?.name && { name: sensorNameInfo.body.name }),
+        ...(sensorDetailsInfo?.body?.name && {
+          name: sensorDetailsInfo.body.name,
+        }),
+        ...(sensorDetailsInfo?.body?.loc && {
+          location: sensorDetailsInfo.body.loc,
+        }),
```

Comment:
It broke the tests when location was not set on a sensor. Pushing a fix.

Reply:
Ah, My bad. I knew I must be missing something. Weeks ago when we first realized this was a problem we should have a made a function called `sanitizeForNextJsProps` or something like that to remove `undefined` or replace them with `null`.  IMHO, the fix should be in a nextjs-sparrow adapter (which I'll be adding in an impending PR) as possibly a function or type (e.g. `sanitizedProps<whatever>`), not in NotehubDataProvider since that's the total wrong end of the program. Basically, I'm trying to think of a way to make it impossible to make this mistake again or forget why we have a `...{}` or `|| null`.

Reply:
I also don't particularly like `null` for some reasons outline in the [long nextjs discussion](https://github.com/vercel/next.js/discussions/11209) about the issue we're hitting. I think a `deleteUndefined` is a better approach or JSON.parse(JSON.serialize(props))

Reply:
I agree with not liking `null`. That caused me some TS errors on the frontend the first time around because I had to make the gateway and sensor models be OK with `string | null`, `number | null`, which just seemed like a lot of extra work. 

I'd prefer to just omit values that come back from Notehub as `undefined` or `null` before the rendering process through `getServerSideProps`.

Reply:
`JSON.parse(JSON.stringify(props))` also opened a whole can of TS worms, as I recall FYI